x/millions/keeper/keeper_pool.go

```diff
@@ -446,6 +452,187 @@ func (k Keeper) UpdatePool(
 	return nil
 }
 
+// RebalanceValidatorsBondings allows rebalancing of validators bonded assets
+// Current implementation:
+// - Initiate an even redelegate distribution from inactive bonded validators to active validators
+func (k Keeper) RebalanceValidatorsBondings(ctx sdk.Context, poolID uint64) error {
+	pool, err := k.GetPool(ctx, poolID)
+	if err != nil {
+		return err
+	}
+
+	// Make sure pool is ready
+	if pool.State == types.PoolState_Created || pool.State == types.PoolState_Unspecified {
+		return types.ErrPoolNotReady
+	}
+
+	_, bondedInactiveVals := pool.BondedValidators()
+	for _, biv := range bondedInactiveVals {
+		if err := k.Redelegate(ctx, pool.PoolId, biv); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Redelegate redistribute evenly the bondedAmount from the bonded inactive to the active valitator set of the pool
+func (k Keeper) Redelegate(ctx sdk.Context, poolID uint64, bondedInactiveVal types.PoolValidator) error {
```

Comment:
Just to clarify.
From what I understand we want to add an additional check inside the `RebalanceValidatorsBondings` before calling `RedelegateToActiveValidators()` to double check valSrcAddr is inactive. As we do already this check by getting `_, bondedInactiveVals := pool.BondedValidators()`, shouldn't we just add this check inside the `RedelegateToActiveValidators` to enable Reusability in case we would call this method somewhere else ?